client/app/scripts/superdesk-authoring/metadata/metadata.js

```diff
@@ -120,7 +119,43 @@ function MetadataCtrl(
         }
     }
 
-    function resolvePublishScheduleDate() {
+    $scope.$watch('item.embargo_date', function(newValue, oldValue) {
+        setEmbargoTS(newValue, oldValue);
+    });
+
+    $scope.$watch('item.embargo_time', function(newValue, oldValue) {
+        setEmbargoTS(newValue, oldValue);
+    });
+
+    /**
+     * Listener method which gets invoked when either Embargo Date or Embargo Time has changed. This function takes
+     * values of both Embargo Date and Embargo Time to form Timestamp.
+     */
+    function setEmbargoTS(newValue, oldValue) {
+        if (newValue !== oldValue) {
+            if ($scope.item.embargo_date && $scope.item.embargo_time) {
+                $scope.item.embargo = datetimeHelper.mergeDateTime(
+                    $scope.item.embargo_date, $scope.item.embargo_time).format();
+            } else {
+                $scope.item.embargo = null;
+            }
+
+            $scope.autosave($scope.item);
+        }
+    }
+
+    /**
+     * Publish Schedule and Embargo are saved as Timestamps in DB but each field has date and time as two different
+     * inputs in UI. This function breaks the timestamp fetched from API to Date and Time and assigns those values to
+     * the appropriate field.
+     */
+    function resolvePublishScheduleAndEmbargoTS() {
+        if ($scope.item.embargo) {
+            var embaroTS = new Date(Date.parse($scope.item.embargo));
```

Comment:
Typo? (embaroTS)

Not that it affects the code itself in any way, though, it's more about about the variable naming...